### DISCUSSION TITLE
help me, I'm stuck with basic Vec2 operation...

### DISCUSSION THREAD

**Discussion by adsick** [bevyengine/bevy#1938] [Q&A]
I have a 
```rust 
struct PointerPosition(Vec2);
```
to store the position of the pointer.
and I want it to be up to date so I try to:
```rust 
fn mouse_move_system(mut mouse_movement: EventReader<MouseMotion>, mut pointer: ResMut<PointerPosition>){
    for mm in  mouse_movement.iter(){
        *pointer.0+=mm.delta;
        println!("pointer position: {:?}", *pointer.0);
    }
}
```
but the catch is that it does not compile!
the error says:
```
binary assignment operation `+=` cannot be applied to type `XY<f32>`
```
what XY\<f32\> and why it does not support addition? what is even going on here? I tried to dig into some glam source code, but there was like 0 documentation on Vec2 and also whole bunch of scary macros.
how to figure it out? I am expecting other newcomers will encounter the same problem.

**Comment by alice-i-cecile** [comment]
Just to double check: does

```rust
        *pointer.0 = pointer.0 + mm.delta;
```
compile?

**Reply by adsick** [reply]
I tried different combinations including what you suggest except I dereferenced the operands. It said "can not add XY\<f32\> to XY\<f32\>"

**Reply by alice-i-cecile** [reply]
That is an even more surprising error message :( If you can't figure this out, I'd recommend filing an issue on the glam repo directly.

**Reply by adsick** [reply]
I'll do it when I will be less lazy than now.

**Comment by adsick** [comment]
Finally I figured it out, guy from Rust Gamedev Russia told me that I do extra dereferences so, just throwing out those asterisks fixed everything. Also noticed that Bevy is unable to start the app if there is no audio available (panic message from rodio)
